app/controllers/v0/sessions_controller.rb

```diff
@@ -113,10 +114,13 @@ def set_cookies
     end
 
     def saml_login_redirect_url(auth: 'success', code: nil)
+      relay_state = params[:RelayState] ? JSON.parse(params[:RelayState]) : {}
       if auth == 'fail'
         url_service.login_redirect_url(auth: 'fail', code: code)
       elsif current_user.loa[:current] < current_user.loa[:highest]
-        url_service.verify_url
+        url_service(relay_state).verify_url
```

Comment:
Don't know if this is necessary... Is it possible for someone who just registered to uplevel? I wouldn't think it is, but want to confirm.

Reply:
If it doesn't ever happen, I think I'd like to refactor URLService to accept query params instead of relay state. And then I can pass the `op=signin` as a param and then set the registration flag in relay state according to that param.